src/dg/dg.cpp

```diff
@@ -1991,13 +2393,15 @@ void DGBase<dim,real,MeshType>::output_results_vtk (const unsigned int cycle)//
 }
 
 template <int dim, typename real, typename MeshType>
-void DGBase<dim,real,MeshType>::allocate_system ()
+void DGBase<dim,real,MeshType>::allocate_system (
+    const bool compute_dRdW, const bool compute_dRdX, const bool compute_d2R)
 {
     pcout << "Allocating DG system and initializing FEValues" << std::endl;
     // This function allocates all the necessary memory to the
     // system matrices and vectors.
 
     dof_handler.distribute_dofs(fe_collection);
+    //This Cuthill_McKee renumbering for dof_handlr uses a lot of memory in 3D, is there another way?
     dealii::DoFRenumbering::Cuthill_McKee(dof_handler,true);
```

Comment:
Is this something we want to change? @dougshidong

Reply:
I welcome other suggestions.

Reply:
How about `DoFRenumbering::boost::king_ordering` ? Seems to be the best choice; see section _A comparison of reordering strategies_ in the link below.

https://www.dealii.org/current/doxygen/deal.II/namespaceDoFRenumbering.html

Reply:
I don't mind. I personally don't have much experience with respect to how ordering affects conditioning. All I know is that Cuthill McKee is a popular and often used ordering scheme in CFD. I doubt it is much less memory-intensive though.

@sivanadarajah might be able to better answer that.

Reply:
> 

Yes RCM is fairly standard. The only time that we had looked other orderings is for Aditya Kashi's work on asynchronous iterations.